### ISSUE TITLE
Possible bug: earthmover() doesn't exit

### ISSUE THREAD

**Issue by zsteve** [devmotion/OptimalTransport.jl#1]
Hey there,

I've tried using `earthmover` to compute OT maps for histograms with square Euclidean cost, however I've encountered a bug that doesn't seem to occur with the unit test examples you have.

I believe a minimal working example is:

```julia
Random.seed!(22)
N = 2; M = N
μ_spt = rand(N)
ν_spt = rand(M)
μ = fill(1/N, N)
ν = fill(1/M, M)
C = pairwise(SqEuclidean(), μ_spt', ν_spt')
@time earthmover(μ, ν, C)
```

For small N, earthmover() behaves properly sometimes, and sometimes crashes. For large N (e.g. N = 10), on my system it seems to crash every time.

Any ideas as to what might be the problem?

**Comment by devmotion** [comment]
Yeah, it's a quite naive and inefficient implementation (basically just taking the explanation in COT right away + combining it with LightGraphs) and was mostly intended for some project I was playing around with. My plan was to implement the much more efficient version of the network simplex algorithm described in Tarjan's paper "Dynamic trees as search trees via Euler tours, applied to the network simplex algorithm" after the NeurIPS deadline.

I'll have a quick look at the issue (maybe it's some convergence check that's missing, due to numerical issues?), but unfortunately I won't spend much time on debugging the issue, since I am going to replace it soon.

**Comment by devmotion** [comment]
I just ran the example above, and it ends up in an infinite loop (so there's definitely some issue :smile:). However, I'm slightly confused about the title of the issue - did you get an error "earthmover() doesn't exist"?

**Comment by zsteve** [comment]
Yep, makes sense! All good.

Re the title: I believe I wrote "earthmover() doesn't _exit_" :)

**Comment by devmotion** [comment]
Oh haha I don't know why I misread the title 😂😂
